### ISSUE TITLE
Slideshow either freezes with black screen on start, or fails to exit cleanly in Virtualbox 5.0

### ISSUE THREAD

**Issue by jlu5** [peterlevi/variety-slideshow#2]
Trying out Variety-slideshow for the first time in a Virtualbox VM running Debian testing/unstable. Sometimes, the slideshow app runs fine, but freezes when exiting (clicking, pressing Escape) on the last frame drawn. Other times, I just get a black screen when running slideshow: Ctrl-C does not help, and I have to go into the virtual console to kill the slideshow program.

Slideshow freezing on exit:

```
gl@midnight:~$ variety-slideshow
/usr/lib/python2.7/dist-packages/varietyslideshow/varietyslideshow.py:19: PyGIWarning: GtkClutter was imported without specifying a version first. Use gi.require_version('GtkClutter', '1.0') before import to ensure that the right version gets loaded.
  from gi.repository import GtkClutter

** (variety-slideshow:2465): WARNING **: Couldn't register with accessibility bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
pci id for fd 9: 80ee:beef, driver (null)
OpenGL Warning: glFlushVertexArrayRangeNV not found in mesa table
OpenGL Warning: glVertexArrayRangeNV not found in mesa table
OpenGL Warning: glCombinerInputNV not found in mesa table
OpenGL Warning: glCombinerOutputNV not found in mesa table
OpenGL Warning: glCombinerParameterfNV not found in mesa table
OpenGL Warning: glCombinerParameterfvNV not found in mesa table
OpenGL Warning: glCombinerParameteriNV not found in mesa table
OpenGL Warning: glCombinerParameterivNV not found in mesa table
OpenGL Warning: glFinalCombinerInputNV not found in mesa table
OpenGL Warning: glGetCombinerInputParameterfvNV not found in mesa table
OpenGL Warning: glGetCombinerInputParameterivNV not found in mesa table
OpenGL Warning: glGetCombinerOutputParameterfvNV not found in mesa table
OpenGL Warning: glGetCombinerOutputParameterivNV not found in mesa table
OpenGL Warning: glGetFinalCombinerInputParameterfvNV not found in mesa table
OpenGL Warning: glGetFinalCombinerInputParameterivNV not found in mesa table
OpenGL Warning: glDeleteFencesNV not found in mesa table
OpenGL Warning: glFinishFenceNV not found in mesa table
OpenGL Warning: glGenFencesNV not found in mesa table
OpenGL Warning: glGetFenceivNV not found in mesa table
OpenGL Warning: glIsFenceNV not found in mesa table
OpenGL Warning: glSetFenceNV not found in mesa table
OpenGL Warning: glTestFenceNV not found in mesa table
libGL error: core dri or dri2 extension not found
libGL error: failed to load driver: vboxvideo
OpenGL Warning: crPixelCopy3D:  simply crMemcpy'ing from srcPtr to dstPtr
^Cgl@midnight:~$
```

Slideshow freezing on start:

```
gl@midnight:~$ variety-slideshow 
/usr/lib/python2.7/dist-packages/varietyslideshow/varietyslideshow.py:19: PyGIWarning: GtkClutter was imported without specifying a version first. Use gi.require_version('GtkClutter', '1.0') before import to ensure that the right version gets loaded.
  from gi.repository import GtkClutter

** (variety-slideshow:1969): WARNING **: Couldn't register with accessibility bus: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
pci id for fd 9: 80ee:beef, driver (null)
OpenGL Warning: glFlushVertexArrayRangeNV not found in mesa table
OpenGL Warning: glVertexArrayRangeNV not found in mesa table
OpenGL Warning: glCombinerInputNV not found in mesa table
OpenGL Warning: glCombinerOutputNV not found in mesa table
OpenGL Warning: glCombinerParameterfNV not found in mesa table
OpenGL Warning: glCombinerParameterfvNV not found in mesa table
OpenGL Warning: glCombinerParameteriNV not found in mesa table
OpenGL Warning: glCombinerParameterivNV not found in mesa table
OpenGL Warning: glFinalCombinerInputNV not found in mesa table
OpenGL Warning: glGetCombinerInputParameterfvNV not found in mesa table
OpenGL Warning: glGetCombinerInputParameterivNV not found in mesa table
OpenGL Warning: glGetCombinerOutputParameterfvNV not found in mesa table
OpenGL Warning: glGetCombinerOutputParameterivNV not found in mesa table
OpenGL Warning: glGetFinalCombinerInputParameterfvNV not found in mesa table
OpenGL Warning: glGetFinalCombinerInputParameterivNV not found in mesa table
OpenGL Warning: glDeleteFencesNV not found in mesa table
OpenGL Warning: glFinishFenceNV not found in mesa table
OpenGL Warning: glGenFencesNV not found in mesa table
OpenGL Warning: glGetFenceivNV not found in mesa table
OpenGL Warning: glIsFenceNV not found in mesa table
OpenGL Warning: glSetFenceNV not found in mesa table
OpenGL Warning: glTestFenceNV not found in mesa table
libGL error: core dri or dri2 extension not found
libGL error: failed to load driver: vboxvideo
OpenGL Warning: crPixelCopy3D:  simply crMemcpy'ing from srcPtr to dstPtr

(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation


(variety-slideshow:1969): Cogl-WARNING **: driver/gl/cogl-framebuffer-gl.c:1186: GL error (1282): Invalid operation
```

**Comment by peterlevi** [comment]
Yes, I also noticed the freezing on exit in Virtualbox on OSX. Unfortunately I have no idea why it happens or what I can do for it. I guess I'll heave to experiment with some simple Python/GTK/Clutter app till I have a contained reproducible case.

**Comment by jlu5** [comment]
This seems fixed in Virtualbox 5.1 :)
